src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,215 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+
+/**
+ * Handles the processing of inputs in the terminal through string processing
+ * @author oliverloo
+ * @version 1.0
+ *
+ */
+public class Parser {
+    /* Takes in an input,
+    1. looks at first word and checks if it is a task
+    2. if it is, splits words future
+     */
+
+    /**
+     * Given an event task string, parses and returns an Event object
+     * @param input
+     * @return event object, formatted with description and time period
+     */
+    public static Event parseEvent(String input) throws DukeException {
+        String event = input.split(" ", 2)[1];
+        String[] event_Arr = event.split(" /from", 2);
+
+        if (event_Arr.length == 2) {
+            String content = event_Arr[0];
+            String[] period_Arr = event_Arr[1].split(" /to");
+            if (period_Arr.length == 2) {
+                String from = period_Arr[0];
+                String to = period_Arr[1];
+                return new Event(content, from, to);
+            } else {
+                throw new DukeException("Invalid Input! You need to specify a /from and /to or content is empty!");
+            }
+        } else {
+            throw new DukeException("Invalid Input! You need to specify a /from and /to or content is empty!");
+        }
+    }
+
+    /**
+     * Given a delete/marked/unmarked command returns the index of that command
+     * @param input
+     * @return returns the index of that command as an int type
+     */
+    public static int getIndex(String input) {
+        String int_Str = input.split(" ", 2)[1];
+        int index = Integer.parseInt(int_Str);
+        return index;
+    }
+
+    /**
+     * Given a deadline task , parses and returns an Deadline object
+     * @param input
+     * @return deadline object, formatted with description and time period
+     */
+    public static Deadline parseDeadline(String input) throws DukeException {
+        String deadline = input.split(" ", 2)[1];
+        String[] deadline_Arr = deadline.split(" /by");
+        if (deadline_Arr.length == 2) {
+            String content = deadline_Arr[0];
+            String date = deadline_Arr[1];
+            return new Deadline(content, date);
+
+        } else {
+            throw new DukeException("Duke.Deadline Input Error! You need to specify date or content is empty!");
+        }
+    }
+
+    /**
+     * Takes a line input extracts the todo task description
+     * @param input
+     * @return description of the todo task
+     */
+    public static String getTodo(String input) {
+        String todo = input.split(" ", 2)[1];
+        return todo;
+    }
+
+    /**
+     * Checks if input is todo task command
+     * @param input
+     * @return true if input is todo task , else false
+     */
+    public static boolean is_toDo(String input) {
```

Comment:
Perhaps omit the underscore here and use camelCase here to adhere to the coding standard!